workflowRDF/fairworkflow.py

```diff
@@ -1,21 +1,81 @@
+import os
 import rdflib
 from rdflib.namespace import RDF, RDFS, DC, XSD, OWL 
 import inspect
 from datetime import datetime
 
-PPLAN = rdflib.Namespace("http://purl.org/net/p-plan/")
-PLEX = rdflib.Namespace("https://plex.org/")
+PPLAN = rdflib.Namespace("http://purl.org/net/p-plan#")
 EDAM = rdflib.Namespace("http://edamontology.org/")
-PROV = rdflib.Namespace("http://www.w3.org/ns/prov/")
+PROV = rdflib.Namespace("http://www.w3.org/ns/prov#")
 DUL = rdflib.Namespace("http://ontologydesignpatterns.org/wiki/Ontology:DOLCE+DnS_Ultralite/")
 BPMN = rdflib.Namespace("https://www.omg.org/spec/BPMN/")
 PWO = rdflib.Namespace("http://purl.org/spar/pwo/")
 RDFG = rdflib.Namespace("http://www.w3.org/2004/03/trix/rdfg-1/")
 NP = rdflib.Namespace("http://www.nanopub.org/nschema#")
 
+
+def nanopublish(assertionrdf=None, uri=None):
```

Comment:
Agreed. This would be much cleaner and potentially easier to reuse in other components of the workbench.

Reply:
So it's not so easy to separate the assertion rdf from the NP since triples in other graphs refer to it. I have instead split the function such that you can generate the nanopub rdf without publishing it.